pkg/client/dtclient/client.go

```diff
@@ -432,25 +435,21 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 	if err != nil {
 		return DynatraceEntity{}, fmt.Errorf("unable to generate external id: %w", err)
 	}
-	settingsWithExternalID, err := d.listSettings(ctx, obj.SchemaId, ListSettingsOptions{
-		Filter: func(object DownloadSettingsObject) bool {
-			return object.ExternalId == legacyExternalID
-		},
-	})
 
+	settingsWithExternalID, err := d.getSettingsFromCache(ctx, obj.SchemaId, func(object DownloadSettingsObject) bool { return object.ExternalId == legacyExternalID })
```

Comment:
My initial thought would have been to make caching a built-in feature of ListSettings and transparent to the callsites. If I just use the public ListSettings now or listSettings in another internal method I'll have to know about and use the FromCache method explicitly. If I use the public interface I have no way to profit from caching.
What made you go for explicitly using the cache?

Reply:
@UnseenWizzard I see the cache as something that will eventually removed again once we do not need to list all settings in upsert anymore. I don't see any other strict/important/useful reason why we need a cache everywhere, but maybe you can help me there :)

Reply:
> once we do not need to list all settings in upsert anymore.
I think with the change to look for unique key properties, this will never go away anymore.

Reply:
I've changed it to be used "built-in" in `listSettings()` :)